tests/beta/queuing/test_beta_operation.py

```diff
@@ -0,0 +1,443 @@
+                ]
+
+    @pytest.mark.parametrize("tensor_observable, expected", tensor_obs)
+    def test_non_identity_obs(self, tensor_observable, expected):
+        """Tests that the non_identity_obs property returns a list that contains no Identity instances."""
+
+        O = tensor_observable
+        for idx, obs in enumerate(O.non_identity_obs):
+            assert type(obs) == type(expected[idx])
+            assert obs.wires == expected[idx].wires
+
+    tensor_obs_pruning = [
+                            (
+                            PauliZ(0) @ Identity(1) @ PauliZ(2),
+                            PauliZ(0) @ PauliZ(2)
+                            ),
+                            (
+                            Identity(0) @ PauliX(1) @ Identity(2) @ PauliZ(3) @  PauliZ(4) @ Identity(5),
+                            PauliX(1) @ PauliZ(3) @ PauliZ(4)
+                            ),
+                            # Single observable is returned
+                            (
+                            PauliZ(0) @ Identity(1),
+                            PauliZ(0)
+                            ),
+                            (
+                            Identity(0) @ PauliX(1) @ Identity(2),
+                            PauliX(1)
+                            ),
+                            (
+                            Identity(0) @ Identity(1),
+                            Identity(0)
+                            ),
+                            (
+                            Identity(0) @ Identity(1),
+                            Identity(0)
+                            ),
+                            (
+                            Identity(0) @ Identity(1) @ Hermitian(herm_matrix, wires=[2,3]),
+                            Hermitian(herm_matrix, wires=[2,3])
+                            )
+                         ]
+
+    @pytest.mark.parametrize("tensor_observable, expected", tensor_obs_pruning)
+    @pytest.mark.parametrize("statistics", [beta_measure.expval, beta_measure.var, beta_measure.sample])
+    def test_prune(self, tensor_observable, expected, statistics):
+        """Tests that the prune method returns the expected Tensor or single non-Tensor Observable."""
+        O = statistics(tensor_observable)
+        O_expected = statistics(expected)
+
+        O_pruned = O.prune()
+        assert type(O_pruned) == type(expected)
+        assert O_pruned.wires == expected.wires
+        assert O_pruned.return_type == O_expected.return_type
+
+    def test_append_annotating_object(self):
+        """Test appending an object that writes annotations when queuing itself"""
+
+        with qml._queuing.AnnotatedQueue() as q:
+            A = PauliZ(0)
+            B = PauliY(1)
+            tensor_op = Tensor(A, B)
+
+        assert q.queue == [A, B, tensor_op]
+        assert q.get_info(A) == {"owner": tensor_op}
+        assert q.get_info(B) == {"owner": tensor_op}
+        assert q.get_info(tensor_op) == {"owns": (A, B)}
```

Comment:
Is this final test the only one that's testing/checking the new behaviour?

Reply:
Is this final test the only one that's testing/checking the new behaviour?

Reply:
So far yes, considering further unit tests involving edge cases.

One open question for me is how we could have the beta `Tensor` class be integrated with the internals of PennyLane. That would be used for test cases involving for example the `@` operator (defined for the `Observable` class on the initial import).

So something along the lines of
```python
A = PauliZ(0)
B = PauliY(1)
C = A @ B
print(type(C))
```
With the output of
```
<class 'pennylane.beta.queuing.operation.BetaTensor'>
```

Reply:
Since `@` is overloaded internally in `Tensor`, aren't all changes self-contained if you're modifying `Tensor`?

Reply:
[`@` is defined for `Observable`](https://github.com/PennyLaneAI/pennylane/blob/61bc47b2b84b6c4cd9483d589a70825d15ebf0b5/pennylane/operation.py#L901), a definition which uses `Tensor`. With the latest approach, `BetaTensor` would subclass `Tensor`, essentially leaving it intact.

Was looking into overriding the import hooks so that a patched `Observable` version could be used in the tests, though would need more exploration.

Reply:
I'd recommend just inheriting all classes that need to be modified to support the new structures. `Observable` will be one of those

Reply:
Alternatively, if `@` is the only issue, I would suggest not even worrying about using it in the tests, simply using `BetaTensor(obs1, obs2)`.

Once we replace the queuing system fully, `@` will naturally start returning the be tensors instead of the old tensors.

Similarly to how, in the qnode refactor, the decorator continued to return old qnodes until the upgrade was complete.

Reply:
`@` is essentially just syntactic sugar, and not an essential part of the queuing process